### PR TITLE
Allow future dates to be entered in DatePicker

### PR DESCRIPTION
**Pull request by luacmartins**
### Details
Coming from [this thread](https://expensify.slack.com/archives/C049HHMV9SM/p1671230484919879), our current Form guidelines state that the user should be able to enter whatever date they want and we should instead let validation handle any errors.

### Fixed Issues
$ https://github.com/Expensify/App/issues/13700  

### Tests
1. Navigate to `Workspace > Connect bank account > Connect manually`
2. Enter `011401533` and `1111222233331111` for routing and account number, respectively. Press `Save & continue`
3. Enter a future date, e.g. `2033-01-01` in the `Incorporation date` input
4. Select another input
5. Verify that you see the `Incorporation date cannot be in the future` error
6. Fill the form with correct data and press `Save & continue`
7. In the `Personal information`, enter a future date for the `Date of birth` input
8. Verify that you see the `Must be over 18 years old` error

- [x] Verify that no errors appear in the JS console

### Offline tests
N/A

### QA Steps
Same as test steps

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

Screenshot 2022-12-19 at 1 01 23 PM

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://user-images.githubusercontent.com/22219519/208512988-d156d714-dcc3-4a23-b8a2-aefb52448051.mov

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://user-images.githubusercontent.com/22219519/208513026-b29d05f4-76ca-4f4a-b210-cffd68ada9d6.mov

</details>

<details>
<summary>Desktop</summary>

https://user-images.githubusercontent.com/22219519/208510888-cabc25a7-0c7e-4a54-85b8-f67197dd4145.mov

</details>

<details>
<summary>iOS</summary>

https://user-images.githubusercontent.com/22219519/208513048-dff7583a-5102-40bb-a9fd-72d65006dc53.mov

</details>

<details>
<summary>Android</summary>

https://user-images.githubusercontent.com/22219519/208513070-8039130c-687c-4823-8789-167b7bfd969d.mov

</details>

### CODE REVIEW

File: src/components/DatePicker/index.js [hunk 1]
```diff
@@ -31,9 +31,6 @@ class DatePicker extends React.Component {
         // Adds nice native datepicker on web/desktop. Not possible to set this through props
         this.inputRef.setAttribute('type', 'date');
         this.inputRef.classList.add('expensify-datepicker');
-        if (this.props.maximumDate) {
-            this.inputRef.setAttribute('max', moment(this.props.maximumDate).format(CONST.DATE.MOMENT_FORMAT_STRING));
-        }
```
Comment (Luke9389) [inline src/components/DatePicker/index.js:36]: These lines lead me to wonder what happens if we don't supply a maximumDate prop. 
Is it necessary to remove the prop all together?
Maybe we could default the behavior to have no maximum date unless one is supplied.

**Review by aimane-chnaif — COMMENTED** [review]
As long as this issue occurs only on iOS, how about leaving it in other platforms?
I understand you made this for platforms consistency but limiting max date is a good feature.

File: src/components/DatePicker/index.js (hunk 1 above)
Comment (luacmartins) [reply src/components/DatePicker/index.js:36]: I'm not sure that I understand the concern. All this line does is also set the `max` attribute for the [type = date](https://github.com/Expensify/App/blob/4f2ef4cf737c8d406c82a656615ba3fe6c16bf19/src/components/DatePicker/index.js#L32) input we set here. The default behavior was already [undefined](https://github.com/Expensify/App/pull/13702/files#diff-33e81c8b58e44ffc9cd1a0528ef9e2c8654f54d2a9a98962ecc2bc85f594b17eL29), but if we never set `maximumDate` there's no need to keep this around.

**Comment by luacmartins** [comment]
@aimane-chnaif as per [this Slack thread](https://expensify.slack.com/archives/C049HHMV9SM/p1671230484919879) our Form guidelines state that we should not limit user input and should instead let validation handle any errors. I think that leaving this prop for other platforms would only introduce inconsistencies and could potentially serve as an example to someone re-introducing this to iOS in the future. 

I think it's best if we remove this altogether.

File: src/components/DatePicker/index.js (hunk 1 above)
Comment (Luke9389) [reply src/components/DatePicker/index.js:36]: Hmm, yeah I suppose that's true. If we're not using it, then we probably should just get rid of it. 
Thx 👍

**Review by Luke9389 — APPROVED** [review]

**Review by aimane-chnaif — APPROVED** [review]
LGTM 🎉 and tests well

**Comment by luacmartins** [comment]
@aimane-chnaif I think those are unrelated to the changes in this PR. We can report them separately. Thanks for the review!

**Comment by yuwenmemon** [comment]
FYI as @mollfpr observed, this PR cause the issue here: https://github.com/Expensify/App/issues/13837

Explanation: https://github.com/Expensify/App/issues/13837#issuecomment-1402888826

**Pull request merged**